packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -75,11 +78,14 @@ const skipRE = /\.(?:map|json)(?:$|\?)/
 export const canSkipImportAnalysis = (id: string): boolean =>
   skipRE.test(id) || isDirectCSSRequest(id)
 
+const inferSelfAccepting = (id: string): false | undefined =>
+  canSkipImportAnalysis(id) ? false : undefined
+
 const optimizedDepChunkRE = /\/chunk-[A-Z\d]{8}\.js/
 const optimizedDepDynamicRE = /-[A-Z\d]{8}\.js/
 
 export function isExplicitImportRequired(url: string): boolean {
-  return !isJSRequest(cleanUrl(url)) && !isCSSRequest(url)
+  return !isJSRequest(url) && !isCSSRequest(url)
```

Comment:
I'm not quite sure how safe it is to change this, but it didn't break any tests (in fact, it fixed a couple tests that were broken by this PR). For example, when a `.html?vue` file has `&lang.js` at the end, we don't want to inject `?import` because that leads to duplicate entries in the module graph. Let me know if this is a problematic solution.